tests/docker/Dockerfile

```diff
@@ -0,0 +1,76 @@
+# PHP Version number as set by offical php docker image tags: https://hub.docker.com/r/library/php/tags/
+# Apache is setup as http server, varnish + nginx is installed for use as reverse proxies
+ARG PHP_VERSION=7.1
+FROM php:${PHP_VERSION}-apache
+
+ENV VARNISH_VERSION=5.1 \
+    VARNISH_MODULES_VERSION=0.12.1 \
+    NGINX_VERSION=1.9.15 \
+    NGINX_CACHE_PURGE_VERSION=2.3 \
+    VARNISH_MALLOC_SIZE="256M" \
+    COMPOSER_ALLOW_SUPERUSER=1 \
+    DEBIAN_FRONTEND=noninteractive
+
+# Use official packages from Varnish and build with varnish-modules mainly for xkey
+# see: https://github.com/varnish/varnish-modules/tree/master/docs
+RUN set -xe \
+        && buildDeps=" \
```

Comment:
is it best practice to chain commands together like this? would it not be better to repeat `RUN`? i am worried how this is for debugging, should one of the setup commands fail.

Reply:
Each `RUN` step creates a new Docker cache layer, so (e.g.) when building something from source and then discarding the source and any `-dev` package dependencies, you want to do that in a single `RUN` step.

Reply:
thanks for the explanation.

Reply:
Basically if this is not done, the image will take longer to build and end up taking lot more space as earlier layers will still contain the files your delete in later layers. It's strange they haven't found a better way to deal with this, like for instance grouping blocks of commands as one layer, but thats a bit outside scope here :)